qiskit/compiler/transpiler.py

```diff
@@ -561,128 +319,144 @@ def _parse_transpile_args(
     # If target is not specified and any hardware constraint object is
     # manually specified then do not use the target from the backend as
     # it is invalidated by a custom basis gate list or a custom coupling map
-    elif basis_gates is None and coupling_map is None:
-        target = _parse_target(backend, target)
+    elif basis_gates or coupling_map:
+        _skip_target = True
+    else:
+        target = getattr(backend, "target", None)
 
-    basis_gates = _parse_basis_gates(basis_gates, backend)
-    initial_layout = _parse_initial_layout(initial_layout, circuits)
-    inst_map = _parse_inst_map(inst_map, backend)
+    initial_layout = _parse_initial_layout(initial_layout)
     coupling_map = _parse_coupling_map(coupling_map, backend)
-    backend_properties = _parse_backend_properties(backend_properties, backend)
-    backend_num_qubits = _parse_backend_num_qubits(backend, num_circuits)
     approximation_degree = _parse_approximation_degree(approximation_degree)
```

Comment:
`_parse_approximation_degree` below still appears to have list-based handling, but I think it's meant to be scalar only.

Reply:
Yeah good catch, we don't need to check the list anymore as that's invalid input now.

Reply:
Removed in: https://github.com/Qiskit/qiskit-terra/pull/10291/commits/75c918b3beaa851c2f121049c2df5365c76f7203